packages/server/services/themeService.ts

```diff
@@ -0,0 +1,137 @@
+import { randomUUID } from 'crypto'
+import type { Model } from 'sequelize-typescript'
+import type { ISiteTheme } from '../models/theme/theme'
+import type { IUserTheme } from '../models/theme/userTheme'
+import { SiteTheme, UserTheme } from '../db'
+
+class ThemeService {
+  createManyThemes = async (themesData: Omit<ISiteTheme, 'uuid'>[]) => {
+    const existingThemes = await SiteTheme.findAll()
+    const existingThemesNames = existingThemes.map(
+      ({ dataValues }) => dataValues.name
+    )
+
+    const themesObjectsList = themesData.flatMap(({ name, description }) => {
+      if (!existingThemesNames.includes(name)) {
+        return {
+          name,
+          description,
+          uuid: randomUUID(),
+        }
+      }
+
+      return []
+    })
```

Comment:
[Предложение]
Мне было бы проще читать запись через `filter`.
Но если тебе больше нравится запись через `flatMap`, оставь ее

```suggestion
    const themesObjectsList = themesData
      .filter(({ name }) => !existingThemesNames.includes(name))
      .map(({ name, description }) => ({
          name,
          description,
          uuid: randomUUID(),
      }));
```